web/static/js/study-responses.js

```diff
@@ -39,14 +39,36 @@ function showResponse(index) {
     $('#response-summary-' + index).show();
 }
 
+function showHideColumns() {
+    // Elements containing links 
+    const show_hide_columns = [{ col: 4, name: "Date" }]
+    const hide_show_elements = show_hide_columns.map(el => {
+        return `<a href class="toggle-vis" data-column="${el.col}">${el.name}</a>`
+    })
+    document.querySelector('.show-hide-cols').innerHTML = `<div class="text-center py-sm-2">Show/Hide: ${hide_show_elements.join(" | ")}</div>`
+
+    // Click event listener on the above links
+    document.querySelectorAll('a.toggle-vis').forEach(el => {
+        el.addEventListener('click', e => {
+            e.preventDefault();
+            const columnIdx = e.target.dataset.column;
+            const column = resp_table.column(columnIdx);
+            column.visible(!column.visible());
+        });
+    })
+
+}
+
 // Datatable init/config for responses table
 const resp_table = $("#individualResponsesTable").DataTable({
     order: [[4, 'desc']], // Sort on "Date" column
     columnDefs: [
-        { className: "column-text-search", targets: [1,2,3] }, // add class to text search columns
+        { className: "column-text-search", targets: [1, 2, 3] }, // add class to text search columns
         { className: "dt-nowrap", "targets": 4 }, // don't wrap "Date" column
         { type: "date", targets: 4 } // set type for "Date" column
     ],
+    bAutoWidth: false, // prevents hide/show cols from growing table
```

Comment:
I think this doesn't matter because the parameter names are backwards-compatible, but FYI, I think `bAutoWidth` has been updated to just `autoWidth` in datatables 1.10+: https://datatables.net/upgrade/1.10-convert